runtime/vm/exceptiondescribe.c

```diff
@@ -219,6 +220,53 @@ printStackTraceEntry(J9VMThread * vmThread, void * voidUserData, UDATA bytecodeO
 	return TRUE;
 }
 
+/**
+* Find the J9Class given a ROMClass.
+* @param[in] vmThread The current VM thread.
+* @param[in] ROMClass The J9ROMClass from which the J9Class is created.
+* @param[in, out] resultClassLoader This is the classLoader that owns the memory segment that the ROMClass was found in.
+*	The romclass memory segment in the shared cache always belongs to vm->systemClassLoader. Set it to the actual class loader which loaded the class in this case.
+*
+* @return The J9class, or NULL on failure.
+*/
+static J9Class*
+findJ9ClassForROMClass(J9VMThread *vmThread, J9ROMClass *romClass, J9ClassLoader **resultClassLoader)
+{
+	J9UTF8 const *utfClassName = J9ROMCLASS_CLASSNAME(romClass);
+	J9JavaVM *vm = vmThread->javaVM;
+	J9Class* ret = NULL;
+	if (j9shr_Query_IsAddressInCache(vm, romClass, romClass->romSize)) {
+		J9ClassLoaderWalkState walkState;
+		J9ClassLoader* classLoader = NULL;
+		BOOLEAN fastMode = J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_FAST_CLASS_HASH_TABLE);
+		if (!fastMode) {
+			omrthread_monitor_enter(vm->classTableMutex);
+		}
+		classLoader = vm->internalVMFunctions->allClassLoadersStartDo(&walkState, vm, 0);
```

Comment:
Will the bootstrap classloader always be iterated first? Otherwise there might be a performance issue.

Reply:
> Will the bootstrap classloader always be iterated first? 

It depends. Bootstrap class loader is the first class loader in the first puddle. If more class loaders have been allocated to the extend that triggers new puddles to be added, the new puddle is added as the new head of the J9Pool. In this case bootstrap class loader won't be the first to be iterated.

Reply:
I suggest checking the bootstrap loader first before iterating through them all, in the hopes it minimizes the perf impact.

Reply:
> I suggest checking the bootstrap loader first before iterating through them all,

I will push the change once the PR builds finish (in case there are more failures).

Reply:
The same as the existing code before this change, I've added code to probe bootstrap class loader as well as app class loader, which further minimizes the chance of iterating all the class loaders.